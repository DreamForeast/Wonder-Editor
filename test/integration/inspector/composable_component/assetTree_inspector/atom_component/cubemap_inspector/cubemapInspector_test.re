open Wonder_jest;

open Expect;

open Expect.Operators;

open NodeAssetType;

open Sinon;

open Js.Promise;

let _ =
  describe("cubemap inspector", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initState(~sandbox, ());
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("prepare currentSelectSource", () => {
      beforeEach(() => {
        MainEditorSceneTool.createDefaultScene(
          sandbox,
          MainEditorAssetTool.initAssetTree,
        );
        CurrentSelectSourceEditorService.setCurrentSelectSource(
          AssetWidgetService.getWidget(),
        )
        |> StateLogicService.getAndSetEditorState;
      });

      describe("test cubemap inspector->show default value", () =>
        test("test snapshot", () => {
          let assetTreeData =
            MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();

          MainEditorAssetChildrenNodeTool.selectCubemapNode(
            ~nodeId=
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                assetTreeData,
              ),
            (),
          );

          BuildComponentTool.buildInspectorComponent(
            TestTool.buildEmptyAppState(),
            InspectorTool.buildFakeAllShowComponentConfig(),
          )
          |> ReactTestTool.createSnapshotAndMatch;
        })
      );

      describe("test cubemap rename", () => {
        describe("test rename to specific name", () =>
          test("test snapshot", () => {
            let assetTreeData =
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();
            let newName = "newCubemapName";

            AssetTreeInspectorTool.Rename.renameAssetCubemapNode(
              ~nodeId=
                MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                  assetTreeData,
                ),
              ~name=newName,
              (),
            );

            BuildComponentTool.buildAssetChildrenNode()
            |> ReactTestTool.createSnapshotAndMatch;
          })
        );

        test(
          {|add cubemap;
            rename cubemap;

            cubemap name should be renamed
              |},
          () => {
            let assetTreeData =
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();
            let newName = "newCubemapToEngine";

            let addedCubemapNodeId = MainEditorAssetIdTool.getNewAssetId();
            MainEditorAssetHeaderOperateNodeTool.addCubemap();

            AssetTreeInspectorTool.Rename.renameAssetCubemapNode(
              ~nodeId=addedCubemapNodeId,
              ~name=newName,
              (),
            );

            MainEditorAssetCubemapNodeTool.getCubemapName(
              ~nodeId=addedCubemapNodeId,
              (),
            )
            |> expect == newName;
          },
        );

        test("if rename to the existed name in the same dir, should fail", () => {
          let assetTreeData =
            MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();

          let cubemap1OldName =
            MainEditorAssetCubemapNodeTool.getCubemapName(
              ~nodeId=
                MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                  assetTreeData,
                ),
              (),
            );

          let addedCubemapNodeId = MainEditorAssetIdTool.getNewAssetId();
          MainEditorAssetHeaderOperateNodeTool.addCubemap();

          let cubemap2OldName =
            MainEditorAssetCubemapNodeTool.getCubemapName(
              ~nodeId=addedCubemapNodeId,
              (),
            );

          AssetTreeInspectorTool.Rename.renameAssetCubemapNode(
            ~nodeId=addedCubemapNodeId,
            ~name=cubemap1OldName,
            (),
          );

          MainEditorAssetCubemapNodeTool.getCubemapName(
            ~nodeId=addedCubemapNodeId,
            (),
          )
          |> expect == cubemap2OldName;
        });
      });

      describe("test load and set face source", () => {
        beforeEach(() => {
          MainEditorAssetTool.buildFakeFileReader();
          MainEditorAssetTool.buildFakeImage();

          LoadTool.buildFakeTextDecoder(LoadTool.convertUint8ArrayToBuffer);
          LoadTool.buildFakeURL(sandbox^);

          LoadTool.buildFakeLoadImage(.);
        });

        testPromise("set source name and src", () => {
          let assetTreeData =
            MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();

          let cubemapTexture =
            MainEditorAssetCubemapNodeTool.getCubemapTextureComponent(
              ~nodeId=
                MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                  assetTreeData,
                ),
              (),
            );

          let imgName = "1.png";
          let imgSrc = "newImgBase64222";

          CubemapInspectorTool.loadAndSetFaceSource(
            ~imgName,
            ~imgSrc,
            ~cubemapTexture,
            ~setSourceFunc=CubemapTextureEngineService.setPXSource,
            (),
          )
          |> then_(_ => {
               let source =
                 CubemapTextureEngineService.unsafeGetPXSource(cubemapTexture)
                 |> StateLogicService.getEngineStateToGetData;

               (Obj.magic(source)##name, Obj.magic(source)##src)
               |> expect == (imgName, imgSrc)
               |> resolve;
             });
        });

        describe("set source format", () => {
          let _test = (imgName, targetFormat) => {
            let assetTreeData =
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();

            let cubemapTexture =
              MainEditorAssetCubemapNodeTool.getCubemapTextureComponent(
                ~nodeId=
                  MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                    assetTreeData,
                  ),
                (),
              );

            CubemapInspectorTool.loadAndSetFaceSource(
              ~imgName,
              ~cubemapTexture,
              ~setSourceFunc=CubemapTextureEngineService.setPXSource,
              (),
            )
            |> then_(_ =>
                 CubemapTextureEngineService.getPXFormat(cubemapTexture)
                 |> StateLogicService.getEngineStateToGetData
                 |> expect == targetFormat
                 |> resolve
               );
          };

          testPromise("set jpg texture to rgb format", () =>
            _test("1.jpg", Wonderjs.TextureType.Rgb)
          );
          testPromise("set jpeg texture to rgb format", () =>
            _test("1.jpeg", Wonderjs.TextureType.Rgb)
          );
          testPromise("set png texture to rgba format", () =>
            _test("1.png", Wonderjs.TextureType.Rgba)
          );
        });

        describe("test snapshot", () => {
          test("should has no face source defaultly", () => {
            let assetTreeData =
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();

            MainEditorAssetChildrenNodeTool.selectCubemapNode(
              ~nodeId=
                MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                  assetTreeData,
                ),
              (),
            );

            BuildComponentTool.buildInspectorComponent(
              TestTool.buildEmptyAppState(),
              InspectorTool.buildFakeAllShowComponentConfig(),
            )
            |> ReactTestTool.createSnapshotAndMatch;
          });
          testPromise("should show face source image after load", () => {
            let assetTreeData =
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();

            let imgSrc = "newImgBase64222";

            CubemapInspectorTool.loadAndSetFaceSource(
              ~imgSrc,
              ~cubemapTexture=
                MainEditorAssetCubemapNodeTool.getCubemapTextureComponent(
                  ~nodeId=
                    MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                      assetTreeData,
                    ),
                  (),
                ),
              (),
            )
            |> then_(_ => {
                 MainEditorAssetChildrenNodeTool.selectCubemapNode(
                   ~nodeId=
                     MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                       assetTreeData,
                     ),
                   (),
                 );

                 BuildComponentTool.buildInspectorComponent(
                   TestTool.buildEmptyAppState(),
                   InspectorTool.buildFakeAllShowComponentConfig(),
                 )
                 |> ReactTestTool.createSnapshotAndMatch
                 |> resolve;
               });
          });
        });
      });

      describe("test change wrap", () =>
        describe("test set wrapS to Repeat", () => {
          let _prepareAndExec = () => {
            let assetTreeData =
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();
            let wrapRepeatType = CubemapInspectorTool.getWrapRepeatType();
            let nodeId =
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                assetTreeData,
              );
            let cubemapTexture =
              MainEditorAssetCubemapNodeTool.getCubemapTextureComponent(
                ~nodeId,
                (),
              );

            MainEditorAssetChildrenNodeTool.selectCubemapNode(~nodeId, ());

            CubemapInspectorTool.changeWrapS(
              ~cubemapTexture,
              ~value=wrapRepeatType,
              (),
            );

            (cubemapTexture, wrapRepeatType);
          };

          test("test snapshot", () => {
            let _ = _prepareAndExec();

            BuildComponentTool.buildInspectorComponent(
              TestTool.buildEmptyAppState(),
              InspectorTool.buildFakeAllShowComponentConfig(),
            )
            |> ReactTestTool.createSnapshotAndMatch;
          });
          test("set wrapS", () => {
            let (cubemapTexture, wrapRepeatType) = _prepareAndExec();

            CubemapTextureEngineService.getWrapS(cubemapTexture)
            |> StateLogicService.getEngineStateToGetData
            |> TextureTypeUtils.convertWrapToInt
            |> expect == wrapRepeatType;
          });

          test("mark need update", () => {
            let (cubemapTexture, wrapRepeatType) = _prepareAndExec();

            let engineState = StateEngineService.unsafeGetState();
            CubemapTextureToolEngine.getIsNeedUpdate(
              cubemapTexture,
              engineState,
            )
            |> expect == true;
          });
        })
      );

      describe("test change filter", () =>
        describe("test set MagFilter", () => {
          let _prepareAndExec = () => {
            let assetTreeData =
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.buildOneCubemapAssetTree();
            let filterNearestType =
              CubemapInspectorTool.getFilterNearestType();
            let nodeId =
              MainEditorAssetTreeTool.BuildAssetTree.Cubemap.getFirstCubemapNodeId(
                assetTreeData,
              );
            let cubemapTexture =
              MainEditorAssetCubemapNodeTool.getCubemapTextureComponent(
                ~nodeId,
                (),
              );

            CubemapInspectorTool.changeMagFilter(
              ~cubemapTexture,
              ~value=filterNearestType,
              (),
            );

            (cubemapTexture, filterNearestType);
          };

          test("set magFilter", () => {
            let (textureComponent, filterNearestType) = _prepareAndExec();

            CubemapTextureEngineService.getMagFilter(textureComponent)
            |> StateLogicService.getEngineStateToGetData
            |> TextureTypeUtils.convertFilterToInt
            |> expect == filterNearestType;
          });
          test("mark need update", () => {
            let (textureComponent, _) = _prepareAndExec();

            let engineState = StateEngineService.unsafeGetState();
            CubemapTextureToolEngine.getIsNeedUpdate(
              textureComponent,
              engineState,
            )
            |> expect == true;
          });
        })
      );
    });
  });