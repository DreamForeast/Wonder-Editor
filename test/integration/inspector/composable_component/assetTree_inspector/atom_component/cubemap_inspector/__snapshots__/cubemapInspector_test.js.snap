// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cubemap inspector prepare currentSelectSource test change wrap test set wrapS to Repeat test snapshot 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-cubemap-inspector"
    >
      <h1>
        Cubemap
      </h1>
      <hr />
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap's name"
        >
          Name
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="cubemap1"
          />
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +X source"
        >
          Right(+ X)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -X source"
        >
          Left(- X)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +Y source"
        >
          Top(+ Y)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -Y source"
        >
          Bottom(- Y)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +Z source"
        >
          Front(+ Z)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -Z source"
        >
          Back(- Z)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="repetitive pattern in horizontal direction of cubemap"
        >
          WrapS
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="2"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="repetitive pattern in vertical direction of cubemap"
        >
          WrapT
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="defines how cubemaps are sampled when a texel covers multiple pixels"
        >
          Mag Filter
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="1"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="defines how cubemaps are sampled when a texel is insufficient to cover a single pixel"
        >
          Min Filter
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
            <option
              value="2"
            >
              Nearest_mipmap_nearest
            </option>
            <option
              value="3"
            >
              Linear_mipmap_nearest
            </option>
            <option
              value="4"
            >
              Nearest_mipmap_linear
            </option>
            <option
              value="5"
            >
              Linear_mipmap_linear
            </option>
          </select>
        </div>
      </article>
    </article>
  </article>
</article>
`;

exports[`cubemap inspector prepare currentSelectSource test cubemap inspector->show default value test snapshot 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-cubemap-inspector"
    >
      <h1>
        Cubemap
      </h1>
      <hr />
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap's name"
        >
          Name
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="cubemap1"
          />
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +X source"
        >
          Right(+ X)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -X source"
        >
          Left(- X)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +Y source"
        >
          Top(+ Y)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -Y source"
        >
          Bottom(- Y)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +Z source"
        >
          Front(+ Z)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -Z source"
        >
          Back(- Z)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="repetitive pattern in horizontal direction of cubemap"
        >
          WrapS
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="repetitive pattern in vertical direction of cubemap"
        >
          WrapT
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="defines how cubemaps are sampled when a texel covers multiple pixels"
        >
          Mag Filter
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="1"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="defines how cubemaps are sampled when a texel is insufficient to cover a single pixel"
        >
          Min Filter
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
            <option
              value="2"
            >
              Nearest_mipmap_nearest
            </option>
            <option
              value="3"
            >
              Linear_mipmap_nearest
            </option>
            <option
              value="4"
            >
              Nearest_mipmap_linear
            </option>
            <option
              value="5"
            >
              Linear_mipmap_linear
            </option>
          </select>
        </div>
      </article>
    </article>
  </article>
</article>
`;

exports[`cubemap inspector prepare currentSelectSource test cubemap rename test rename to specific name test snapshot 1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/cubemap.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        newCubemapName
      </span>
    </div>
  </article>
</article>
`;

exports[`cubemap inspector prepare currentSelectSource test load and set face source test snapshot should has no face source defaultly 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-cubemap-inspector"
    >
      <h1>
        Cubemap
      </h1>
      <hr />
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap's name"
        >
          Name
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="cubemap1"
          />
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +X source"
        >
          Right(+ X)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -X source"
        >
          Left(- X)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +Y source"
        >
          Top(+ Y)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -Y source"
        >
          Bottom(- Y)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +Z source"
        >
          Front(+ Z)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -Z source"
        >
          Back(- Z)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="repetitive pattern in horizontal direction of cubemap"
        >
          WrapS
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="repetitive pattern in vertical direction of cubemap"
        >
          WrapT
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="defines how cubemaps are sampled when a texel covers multiple pixels"
        >
          Mag Filter
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="1"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="defines how cubemaps are sampled when a texel is insufficient to cover a single pixel"
        >
          Min Filter
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
            <option
              value="2"
            >
              Nearest_mipmap_nearest
            </option>
            <option
              value="3"
            >
              Linear_mipmap_nearest
            </option>
            <option
              value="4"
            >
              Nearest_mipmap_linear
            </option>
            <option
              value="5"
            >
              Linear_mipmap_linear
            </option>
          </select>
        </div>
      </article>
    </article>
  </article>
</article>
`;

exports[`cubemap inspector prepare currentSelectSource test load and set face source test snapshot should show face source image after load 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-cubemap-inspector"
    >
      <h1>
        Cubemap
      </h1>
      <hr />
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap's name"
        >
          Name
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="cubemap1"
          />
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +X source"
        >
          Right(+ X)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <img
              src="newImgBase64222"
            />
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -X source"
        >
          Left(- X)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +Y source"
        >
          Top(+ Y)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -Y source"
        >
          Bottom(- Y)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap +Z source"
        >
          Front(+ Z)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="cubemap -Z source"
        >
          Back(- Z)
        </div>
        <div
          className="item-content item-texture"
        >
          <div
            className="texture-img"
          >
            <input
              className="cubemap-faceSource-loadAndSetSource"
              multiple={false}
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="repetitive pattern in horizontal direction of cubemap"
        >
          WrapS
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="repetitive pattern in vertical direction of cubemap"
        >
          WrapT
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="defines how cubemaps are sampled when a texel covers multiple pixels"
        >
          Mag Filter
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="1"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="defines how cubemaps are sampled when a texel is insufficient to cover a single pixel"
        >
          Min Filter
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
            <option
              value="2"
            >
              Nearest_mipmap_nearest
            </option>
            <option
              value="3"
            >
              Linear_mipmap_nearest
            </option>
            <option
              value="4"
            >
              Nearest_mipmap_linear
            </option>
            <option
              value="5"
            >
              Linear_mipmap_linear
            </option>
          </select>
        </div>
      </article>
    </article>
  </article>
</article>
`;
